src/codegen/function_builder.cpp

```diff
@@ -131,10 +131,13 @@ FunctionBuilder::FunctionBuilder(
           cc,
           ConstructFunction(cc, name, FunctionDeclaration::Visibility::External,
                             ret_type, args)) {}
-                            
+
 FunctionBuilder::~FunctionBuilder() {
   if (!finished_) {
-    throw Exception{"Missing call to FunctionBuilder::ReturnAndFinish()"};
+    LOG_ERROR(
+        "Missing call to FunctionBuilder::ReturnAndFinish() for function '%s'",
+        func_->getName().data());
+    PELOTON_ASSERT(false);
```

Comment:
On second thought, let's remove this assertion. In `Debug` mode, the destructor gets called on the way up through the call stack when an exception is thrown. We don't want this assertion to trigger here and potentially hide any exception context.